pkg/httpserver/httpserver.go

```diff
@@ -53,7 +54,15 @@ func New(opts *Options, logger *zap.SugaredLogger) (*Server, error) {
 		return nil, err
 	}
 
-	handler, err := newHandlerFromConfig(config, logger)
+	notFoundHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		logger.Debugf("request did not match with any rule: %s", strRequest(r))
+		w.WriteHeader(404)
+		if opts.ExitOnUnmatchedRule {
+			logger.Fatalf("--exit-on-unmatched-rule is set, exiting")
```

Comment:
I think it would be more graceful to asynchronously stop the server. With os.Exit being called from the handler there is no guarantee that the response has been flushed to the client.